applications/HDF5Application/python_scripts/single_mesh_primal_input_process.py

```diff
@@ -11,18 +11,27 @@ def Factory(settings, Model):
                 "model_part_name" : "MainModelPart",
                 "file_settings" : {},
                 "nodal_results_settings" : {},
-                "time_tag_precision" : 4
+                "element_results_settings" : {},
+                "time_tag_precision" : 4,
+                "file_name": "DEFAULT_NAME"
             }
             """)
     settings = settings["Parameters"].Clone()
     settings.ValidateAndAssignDefaults(default_settings)
     model_part = Model[settings["model_part_name"].GetString()]
     hdf5_file_factory = hdf5_io.HDF5SerialFileFactory(settings["file_settings"])
     nodal_results_input = hdf5_io.PrimalBossakInput(settings["nodal_results_settings"])
+    element_results_input = hdf5_io.ElementResultsInput(settings["element_results_settings"])
     input_time_settings = KratosMultiphysics.Parameters("""{}""")
     input_time_settings.AddEmptyValue("time_tag_precision")
+    input_time_settings.AddEmptyValue("file_name")
+    if settings["file_name"].GetString()=="DEFAULT_NAME":
```

Comment:
Same here.

Reply:
The problem is, model_part is initialized after ValidateAndAssignDefaults, therefore I cannot use it inside the default_settings.

Reply:
Ok, I see now.